### ISSUE TITLE
Documentation / Tutorials?

### ISSUE THREAD

**Issue by Tobiasartz** [go-swagger/go-swagger#559]
I've noticed that there is very little documentation on how to generate swagger files from annotations.
Is this really all there is http://goswagger.io/generate/spec/?

How about some code examples that show annotations being used?

**Comment by casualjim** [comment]
There is the examples folder in the repo, all generated code also uses all the annotations that are applicable for it.
https://github.com/go-swagger/go-swagger/tree/master/examples/todo-list

This is the code I used to test parsing the annotations: https://github.com/go-swagger/go-swagger/tree/master/fixtures/goparsing/classification

Please bear in mind that this is a project (not a product!) to which others and I have contributed significant amounts of free time to get it to where it is today. All help we can get is absolutely welcome. 

To make things better:
The site source is here, uses hugo. Feel free to submit a PR with some more examples; I'll gladly merge

**Comment by martinShil** [comment]
@casualjim I used your example [test parsing the annotations](https://github.com/go-swagger/go-swagger/tree/master/fixtures/goparsing/classification) and use a command 
`swagger generate spec -o ./swagger.json`

And again I got an empty definitions fields in a json file. I tried to use it in my own project, but again it parses the main file and doesn't go deeply (doesn't scan other packages), but in doc we have.

> You give it a main file and it will parse all the files that are reachable by that main package to produce a swagger specification.

**Comment by nathany** [comment]
While I agree with the project vs. product sentiment, it's difficult to help document something I don't understand. 😕

**Comment by GlenDC** [comment]
@nathany the latest documentation for the code on master can be found here: https://github.com/go-swagger/go-swagger/tree/master/docs/generate/spec

Improvement of documentation is always a good request. You are in a unique position however as you are a newcomer and thus have the perspective that is perfectly fit to help with the improvement of documentation.

Could you please add questions here that are not answered by the documentation. Also any suggestions on what according to you is missing, is always more then welcome :)

**Comment by nathany** [comment]
@GlenDC Thanks for the link. I see swagger:operation is documented there but not yet on https://goswagger.io.

Right now I'm having some trouble getting the `Responses:` section working for swagger:router, including some routes that have just a status code with no other response. While searching around the code, I also saw `description:` for the route responses. There seems to be some inconsistency between what's documented and the examples lying around, which makes me less certain why validation is failing (also #859).

What I think would help the most is a tutorial that walks through generating swagger.json from code annotations, assuming someone is coming from an existing code base (even if it were a to do app).

**Comment by nathany** [comment]
I've decided to write swagger.yml instead of using the code annotations for now. It's going well.

**Comment by GlenDC** [comment]
@nathany yes that's the recommended and more sane approach. Also saves you tons of work. Also in the end writing annotations doesn't save that much work compared to writing the spec, as you end up writing most of the stuff anyhow.

None the less, does not mean that annotations should not be well documented. Your tutorial suggestion is definitely a good idea.

Could you please be a bit more clear about the other stuff? Please be as clear as possible with the questions and to what they relate. This way we can take it into account when improving the documentation.

**Comment by nathany** [comment]
Regarding the other questions, in the `swagger.yml` I just write something like this:

```
responses:
        201:
          description: Created
        400:
          description: Bad request
          schema:
            $ref: "#/definitions/error"
        401:
          description: Unauthorized
        500:
          description: Unexpected error
          schema:
            $ref: "#/definitions/error"
```

But I was having trouble getting the equivalent to work as code annotations. Perhaps I just wasn't formatting it correctly?
